### PR TITLE
Added removeAllSounds method to the unload method in the Sound module…

### PR DESCRIPTION
**Pull request by greenfngrs**
… to fix audio memory leak.

### CODE REVIEW

**Comment by greenfngrs** [comment]
@bigtimebuddy, @andrewstart, @aberkie, @brianmlink: Take a look!

**Comment by brianmlink** [comment]
LGTM Jer

File: src/media/Sound.js [hunk 1]
```diff
@@ -1517,6 +1517,9 @@
 			}
 			SoundJS.removeSound(list[i]);
 		}
+
+		//This is to release the audio binary that remains in the jsArrayBufferData.
+		SoundJS.removeAllSounds();
```
Comment (andrewstart) [inline src/media/Sound.js]: We probably don't want to unload all sounds here - removeAllSounds() could be called in unloadAll(), but here we want to only unload the list items, and nothing else - we wouldn't want to try to play a sound that we think is loaded, but SoundJS does not.

**Comment by greenfngrs** [comment]
Won't this change actually unload all sounds, even when the caller's intent is to only unload a specified list of sounds?

**Comment by greenfngrs** [comment]
Just saw Andrew's comment. Agree. This needs more work. thanks!

**Comment by yanshcherbakovatredspace** [comment]
@andrewstart Thank you for pointing this out. I was anticipating an oversight on my part that it might not have been a good idea to place that in the unload function rather the unloadAll function. I have made this change.

**Comment by andrewstart** [comment]
Does this actually clean up memory where `unload()` was not? `SoundJS.removeSound(list[i]);` in `unload()` is the same function that `SoundJS.removeAllSounds()` is calling, but we are just passing the id that we created the sound with instead of the src (the function says it supports that). If that isn't working, then we could pass the src instead, as we also keep track of that.

**Comment by yanshcherbakovatredspace** [comment]
@andrewstart The odd thing is that `unload()` does what is intended when Springroll is used with Pixi, as in the memory frees up but this doesn't happen when used with EaselJS.

I will test passing the src.

**Comment by andrewstart** [comment]
How did passing the src work out?

**Comment by yanshcherbakovatredspace** [comment]
@andrewstart Passing the src works perfectly.

**Pull request merged**